src/lib/meta/meta_blk_service.cpp

```diff
@@ -375,7 +377,12 @@ void MetaBlkService::register_handler(meta_sub_type type, const meta_blk_found_c
     m_sub_info[type].cb = cb;
     m_sub_info[type].comp_cb = comp_cb;
     m_sub_info[type].do_crc = do_crc ? 1 : 0;
-    HS_LOG(INFO, metablk, "[type={}] registered with do_crc: {}", type, do_crc);
+    m_dependency_topological_graph[type];
```

Comment:
You are trying to put every client into the graph and during recover do a few for loops to determin independent clients and then erase from the graph, which is unncessary. 

We already have a client_map, m_sub_info, which contains every client's private information and put a new field: bool no_dep {true}; 

during register_handler, if this client appears in anyone's dep vector, set this client's no_dep to false, your graph's existing operation stays the same. 

During recovery, loop this m_sub_info, if no_dep is true, call recover_meta_blk(m_sub_info(client).meta_blk). 
Then loop your graph to call recover_meta_blk() in order.

Reply:
you cannot define yourself as no_dep as you only know you dont depends on anyone, but you dont know if there is anyone depends on you.

Reply:
>during register_handler, if this client appears in anyone's dep vector, set this client's no_dep to false, your graph's existing operation stays the same.

a subsystem can be excluded from the graph only if it depends on none, and none depends on it too. if this client does not appear in anyone's dep vector, it means none depends on it. but you have no knowledge whether it depends on anyone.

Reply:
It is deduced after all register completes. For every resigter_hander, for this clicent, and every client in dep vec, set m_sub_info[client].no_dep = false. It can't be determined by one register call.